src/Service/LearningMaterialTextExtractor.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Entity\DTO\LearningMaterialDTO;
+use App\Exception\LearningMaterialTextExtractorException;
+use Exception;
+use Vaites\ApacheTika\Client;
+
+class LearningMaterialTextExtractor
+{
+    private bool $enabled = false;
+    private Client $client;
+
+    public function __construct(
+        protected NonCachingIliosFileSystem $fileSystem,
+        protected TemporaryFileSystem $temporaryFileSystem,
+        protected IliosFileSystem $iliosFileSystem,
+        ?Client $client = null,
+    ) {
+        if ($client) {
+            $this->enabled = true;
+            $this->client = $client;
+        }
+    }
+
+    public function extract(LearningMaterialDTO $dto): void
+    {
+        if (!$this->enabled) {
+            return;
+        }
+
+        if (!$dto->filename) {
+            //not a File LM
+            return;
+        }
+
+        if (!$this->client->isMIMETypeSupported($dto->mimetype)) {
+            //not the type of file tika can extract
+            return;
+        }
+
+        if ($this->iliosFileSystem->checkIfLearningMaterialTextFileExists($dto->relativePath)) {
+            //this LM has already been extracted
+            return;
+        }
+
+        if (!$this->fileSystem->checkLearningMaterialRelativePath($dto->relativePath)) {
+            throw new LearningMaterialTextExtractorException(
+                "There is no material on this system at $dto->relativePath"
+            );
+        }
+
+        $contents = $this->fileSystem->getFileContents($dto->relativePath);
+        $tmpFile = $this->temporaryFileSystem->createFile($contents);
+
+        try {
+            $text = $this->client->getText($tmpFile->getRealPath());
+            $this->iliosFileSystem->storeLearningMaterialText($dto->relativePath, $text);
+        } catch (Exception $exception) {
+            if (
+                $exception->getCode() === 422 &&
```

Comment:
suggesting using  `Symfony\Component\HttpFoundation\Response::HTTP_UNPROCESSABLE_ENTITY` instead of `422` here.

Reply:
Interesting. The error is coming from [hard coding](https://github.com/vaites/php-apache-tika/blob/792dc1254b4ccd92c6964ab477a1626dca6784ee/src/Clients/WebClient.php#L620) in the tika library. I wasn't thrilled with that. I'm worried using a constant would obscure this further. But the constant is 422 so I'm not opposed. I'm really just leaving a comment for posterity, I'll make the change and combine it with a better comment on the origin of that error code.